lib/common/view/tagPopup.dart

```diff
@@ -0,0 +1,108 @@
+import "package:flutter/material.dart";
+import 'package:flutter_colorpicker/flutter_colorpicker.dart';
+import 'dart:async';
+
+import '../../models/tag.dart';
+import '../../view/loginView.dart';
+import 'flashError.dart';
+
+/// Display a popup to select a tag with a colorpicker. Takes in the runtime context and an optional setState function to call on update for updating a view as necessary.
+
+Future<List<Tag>> showTagSelectionDialog(BuildContext context,
+    {Function? setState}) async {
+  List<Tag> selectedTags = [];
+
+  TextEditingController nameController = TextEditingController();
+  Color selectedColor = Colors.blue;
+  Color pickerColor = const Color(0xff443a49);
+
+  void changeColor(Color color) {
+    pickerColor = color;
+    selectedColor = color;
+    if (setState != null) {
+      setState(() {});
+    }
+  }
+
+  await showDialog(
```

Comment:
dart doesnt like doing anything with buildcontext in async, so this will show squiggly blue lines on all IDEs probably. not sure why it didnt show before, but theyre probably right